Mathlib/Data/Set/Finite/Basic.lean

```diff
@@ -669,31 +669,39 @@ theorem finite_option {s : Set (Option α)} : s.Finite ↔ { x : α | some x ∈
     ((h.image some).insert none).subset fun x =>
       x.casesOn (fun _ => Or.inl rfl) fun _ hx => Or.inr <| mem_image_of_mem _ hx⟩
 
+/-- Induction principle for finite sets: To prove a property `C` of a finite set `s`, it's enough
+to prove for the empty set and to prove that `C t → C ({a} ∪ t)` for all `t`.
+
+See also `Set.Finite.induction_on` for the version requiring to check `C t → C ({a} ∪ t)` only for
+`t ⊆ s`. -/
 @[elab_as_elim]
-theorem Finite.induction_on {C : Set α → Prop} {s : Set α} (h : s.Finite) (H0 : C ∅)
-    (H1 : ∀ {a s}, a ∉ s → Set.Finite s → C s → C (insert a s)) : C s := by
-  lift s to Finset α using h
-  induction' s using Finset.cons_induction_on with a s ha hs
-  · rwa [Finset.coe_empty]
-  · rw [Finset.coe_cons]
-    exact @H1 a s ha (Set.toFinite _) hs
-
-/-- Analogous to `Finset.induction_on'`. -/
+theorem Finite.induction_on {C : ∀ s : Set α, s.Finite → Prop} (s : Set α) (hs : s.Finite)
```

Comment:
Can we rename `C` to `motive`, to match the convention set by builtin induction principles?